### DISCUSSION TITLE
How to call mimc to input a string and output the mimc hashed result?

### DISCUSSION THREAD

**Discussion by fengyin450** [Consensys/gnark-crypto#444] [Q&A]
In the example of [https://play.gnark.io/#](url), it can be seen that the hash result of "Secret": "0xdeadf00d" is "Hash": "1037254799353855871006189384309576393135431139055333626960622147300727 796413", how to call [https://github.com/Consensys/gnark-crypto/blob/master/ecc/bn254/fr/mimc/mimc.go](url) can input Secret to get Hash by implementing mimc function? I tried many methods, but the output is 0 or wrong result. The following is one of them:
```

func mimcHash(data []byte) string {

	f := bn254.NewMiMC()

	f.Write(data)

	hash := f.Sum(nil)

	hashInt := big.NewInt(0).SetBytes(hash)

	return hashInt.String()

}
 
```

**Comment by fengyin450** [comment]
@AlexandreBelling @Tabaie 😭Can someone kindly help me?

**Comment by ivokub** [comment]
Answered in https://github.com/Consensys/gnark-crypto/discussions/443
